lib/mail/message.rb

```diff
@@ -249,13 +249,14 @@ def inform_interceptors
     #  mail.deliver
     def deliver
       inform_interceptors
-      if delivery_handler
+      response = if delivery_handler
         delivery_handler.deliver_mail(self) { do_delivery }
       else
         do_delivery
       end
       inform_observers
-      self
+      
+      response
```

Comment:
I just return response because it seems `delivery_method.settings[:return_response]` is checked in each specific `delivery_method.deliver!` implementation